src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -201,17 +216,24 @@ public void AddReceiver(string eventHubName, string receiverConnectionString, st
             };
         }
         
-        internal EventHubClient GetEventHubClient(string eventHubName)
+        internal EventHubClient GetEventHubClient(string eventHubName, string connection)
         {
             EventHubClient client;
             if (_senders.TryGetValue(eventHubName, out client))             
             {
                 return client;
             }
+            else if (!string.IsNullOrWhiteSpace(connection) && NameResolver != null)
+            {
+                var resolvedConnection = NameResolver.Resolve(connection);
```

Comment:
In the IConnectionProvider implementation, the Connection property is marked as [AppSetting] so you shouldn't need to resolve this on your own?